test/functorch/test_aotdispatch.py

```diff
@@ -3490,6 +3532,21 @@ def f(a, b):
         self.assertEqual(a_ref_base.grad.a, a_test_base.grad.a)
         self.assertEqual(a_ref_base.grad.b, a_test_base.grad.b)
 
+    def test_traceable_subclass_with_differently_sized_inner_tensor(self):
+        data = torch.rand(2, 4)
+        scale = torch.rand(1)
+        subclass = ScaledTensor(data, scale)
+
+        def func(tensor_a, tensor_b):
+            return torch.add(tensor_a, tensor_b)
+
+        add_tens = torch.rand(2, 4)
+        real_out = func(data, add_tens)
+        func = torch.compile(func, backend="eager")
```

Comment:
Since this test lives in `test_aotdispatch.py`, can you update it to use the `aot_eager` backend?

Reply:
I think it would also be good to have some tests for:

(1) `dynamic=True`, run the compiled fn multiple times with different input shapes and don't see any recompiles (the test above isn't dynamic, I think?)

(2) testing the default behavior of mark_dynamic with subclasses: maybe have a subclass with two inner tensors, one with the same # dims as the subclass (`x.a`), and one with a different # of dims (`x.b`). And then if we mark_dynamic the subclass, we should see that we don't recompile if `x.a`'s shape changes, but we do recompile if `x.b`'s shape changes.

LMK what you think

(2) manually calling `mark_dynamic()` on a tensor subclass, making sure that we don't recompile (e.g. that the inner tensor is properly marked dynamic

Reply:
Added tests and moved the old one into `test/dynamo/test_subclasses.py`